backend-tests/tests/test_access.py

```diff
@@ -0,0 +1,418 @@
+        tadmm = ApiClient(tenantadm_v2.URL_MGMT)
+
+        res = tadmm.with_auth(tenant.auth).call(
+            "POST",
+            tenantadm_v2.URL_TENANT_UPGRADE_START,
+            path_params={"id": tenant.id},
+        )
+
+        assert res.status_code == 200
+        res = res.json()
+
+        stripeutils.confirm("pm_card_visa", res["intent_id"])
+
+        body = {
+            "plan": "enterprise",
+        }
+
+        res = tadmm.with_auth(tenant.auth).call(
+            "POST",
+            tenantadm_v2.URL_TENANT_UPGRADE_COMPLETE,
+            path_params={"id": tenant.id},
+            body=body,
+        )
+        assert res.status_code == 202
+
+        update_tenant(
+            tenant.id, addons=["troubleshoot", "configure"],
+        )
+
+        self.check_access_remote_term(tenant.auth, tenant.device_id)
+        self.check_access_file_transfer(tenant.auth, tenant.device_id)
+        self.check_access_deviceconfig(tenant.auth, tenant.device_id)
+        self.check_access_auditlogs(tenant.auth)
+        self.check_access_sessionlogs(tenant.auth)
+        self.check_access_rbac(tenant.auth)
+
+
+def _make_tenant(plan):
+    uuidv4 = str(uuid.uuid4())
+    tenant, username, password = (
+        "test.mender.io-" + uuidv4,
+        "ci.email.tests+" + uuidv4 + "@mender.io",
+        "secretsecret",
+    )
+    email = "ci.email.tests+" + uuidv4 + "-user2@mender.io"
+
+    # Create tenant with two users
+    tenant = create_org(tenant, username, password, plan=plan)
+    tenant.users.append(create_user(email, password, tenant.id))
+    update_tenant(
+        tenant.id, addons=[],
+    )
+    r = ApiClient(useradm.URL_MGMT).call(
+        "POST", useradm.URL_LOGIN, auth=(email, password),
+    )
+    assert r.status_code == 200
+    tenant.auth = r.text
+    tenant.device_id = str(uuid.uuid4())
+    return tenant
+
+
+def _make_trial_tenant():
+    uuidv4 = str(uuid.uuid4())
+    tname = "test.mender.io-{}-{}".format(uuidv4, "trial")
+    email = "some.user+{}@example.com".format(uuidv4)
+    password = "correcthorse"
+
+    tadmm = ApiClient(tenantadm_v2.URL_MGMT)
+
+    args = {
+        "organization": tname,
+        "email": email,
+        "password": password,
+        "name": "foo",
+        "g-recaptcha-response": "dummy",
+        "plan": "enterprise",
+    }
+
+    res = tadmm.call("POST", tenantadm_v2.URL_CREATE_ORG_TRIAL, body=args,)
+
+    assert res.status_code == 202
+
+    tadmi = ApiClient(tenantadm.URL_INTERNAL, host=tenantadm.HOST, schema="http://")
+
+    res = tadmi.call(
+        "GET",
+        tenantadm.URL_INTERNAL_TENANTS,
+        qs_params={"username": email},  # urllib.parse.quote(email)}
+    )
+
+    assert res.status_code == 200
+    assert len(res.json()) == 1
+
+    api_tenant = res.json()[0]
+    cli = CliTenantadm()
+    tenant = cli.get_tenant(api_tenant["id"])
+    tenant = json.loads(tenant)
+    tenant_token = tenant["tenant_token"]
+
+    propagate_wait_s = 2
+    max_tries = 255
```

Comment:
8 minutes and a half seem a bit conservative, maybe 5 minutes will be enough?
just asking, 5 minutes is also finger in the air guess ;)

Reply:
fixed!
I trust your fingers!